fix(content): Yet another pirate job bugfix

ziproot opened the pull request:
**Bug fix**

This PR addresses the bugs described on Discord [[1]](https://discord.com/channels/251118043411775489/536900466655887360/1201722778999328778) [[2]](https://discord.com/channels/251118043411775489/1136144420681560064/1226327781533679687)

## Summary
I found several bugs with pirate jobs:
- Only the eliminating competition jobs have the "marked" personality trait, so almost all of them allow you to just wait for NPCs to do the job for you (though for the merchant attack ones this is intentional)
**FIX: This PR gives them the marked personality trait.**
- Several jobs ask you to travel from a CCOR planet to Men, or vice versa, asking you to escort through lawful space, or smuggle goods, slaves, etc. (CCOR doesn't fine you or confiscate the goods), or eliminate a militia fleet in CCOR space (which, due to the first listed bug, means pirates will almost immediately do the job for you), or assasinate a free worlds politican who decided to enter CCOR space (not as trivial because fw ships are better armed), or attack a merchant convoy that entered CCOR space [CCOR space only has pirates and ccor ships normally]. Basically, you shouldn't be offered any pirate jobs between CCOR systems that imply you should be careful about getting scanned, and you shouldn't be offered any "attack" jobs where NPCs are within CCOR space unless the NPCs are pirates or ccor
**FIX: This pr doesn't offer those jobs in ccor planets, and destinations are not ccor planets, and npcs do not spawn in ccor planets in case the mission is given in Men.**
- There is no penalty in terms of pirate reputation for aborting a slave job
**FIX: I added a penalty along with a generic phrase associated with it.**

During the April 2024 Campfire Chat, @JustSomeDude found an additional bug:
- You can get fined for drug running and shady passenger missions when landing at the destination
**FIX: Gave those missions infiltrating. I have no idea why they didn’t have that already.**

I found four other bugs that I cannot fix without code PRs or regressions:
- You cannot get caught transporting slaves or transporting shady passengers because fines don’t work on missions that don’t involve cargo (thanks @warp-core)
**FIX: Can be done better with a code PR, and for the slavery missions this will prevent major regressions like the Quarg claiming you have Quarg tech when you “only” have slaves.**
- The slave jobs are not failed if you are sentenced to death, even if this happens in space where you can survive, and use the default message
**FIX: Not possible with atrocity failures, requires a code PR.**
- If you get caught with atrocity level cargo, the default atrocity message includes the fail message for other missions inside of it
**FIX: Requires a code PR.**
- Being sentenced to death for any reason leads to the Quarg death sentence mentioning Quarg technology, which prevents you from being able to have anything else give a death sentence
**FIX: Requires a code PR.**
## Testing Done
Made sure I didn't get jobs to/from ccor systems. Also made sure NPCs spawned not in CCOR systems. Made sure NPCs weren't taken out by non-player ships except for merchants.

Tested the abort penalty for slavery missions.

Tested the drug running and shady passenger missions. Was no longer fined when landing on the destination planet.

## Save File
[Remnant Test~Pirate Test.txt](https://github.com/endless-sky/endless-sky/files/14092419/Remnant.Test.Pirate.Test.txt)
[Remnant Test~Drug Running.txt](https://github.com/endless-sky/endless-sky/files/14952224/Remnant.Test.Drug.Running.txt)

ziproot commented:
The integration test is failing because it's stupid and thinks atrocities don't have to be children of cargo nodes

Saugia commented:
Any reason why you're removing a contributor in this PR?

Nevermind, I see the commit name, but do include that in the description so people know what's going on there.

ziproot commented:
The scanning message for death sentences is completely broken and can show the message for a different job. It also integrates the message into a custom one (that I can fix). The former, however, is a bug that I'm stuck with, so I'm marking this as draft until that bug is addressed.

ziproot commented:
This only partially addresses the issues (it does as much as possible with data only edits). I will attempt to look at the code, but I am still learning the Endless Sky codebase.

ziproot commented:
![image](https://github.com/endless-sky/endless-sky/assets/109186806/d3534499-b284-43c1-9a0f-420bad857d09)
This was from me landing on a quarg planet with a transport slaves mission active. Marking as draft until a resolution is found.

ziproot commented:
I’m considering deleting the slave food part as that’s where most of the issues come in, and then all of the problems with that can be fixed in a separate code PR:
-atrocity fines being able to be their own line and not a child of a cargo node
-having the atrocity fine message override the default/faction death sentence message
-have jobs with atrocity level fines fail if you are given the death sentence in space

ziproot commented:
Converting to draft as my PRs are still being merged before testing is finished

warp-core commented on data/human/pirate jobs.txt:
```diff
@@ -1691,6 +1697,7 @@ mission "Slave Transport [1]"
 	repeat
 	description "A pirate on <planet> has bought slaves from a local warlord. Transfer these <bunks> slaves in secure cells to <destination> for <payment>. Slavery is punishable by death throughout the galaxy, so this may be a dangerous job to take."
 	passengers 2 10 .9
+	# TODO: illegal -1 should be an atrocity, but it isn't
```
```suggestion
```

warp-core commented on data/human/pirate jobs.txt:
```diff
@@ -1715,6 +1727,7 @@ mission "Slave Transport [2]"
 	repeat
 	description "A pirate on <planet> has bought slaves from a local warlord. Transfer these <bunks> slaves in secure cells to <destination> for <payment>. Slavery is punishable by death throughout the galaxy, so this may be a dangerous job to take."
 	passengers 2 10 .9
+	# TODO: illegal -1 should be an atrocity, but it isn't
```
```suggestion
```

warp-core commented on data/human/pirate jobs.txt:
```diff
@@ -1741,6 +1759,7 @@ mission "Bulk Slave Transport [0]"
 	repeat
 	description "A pirate on <planet> has bought slaves from a local warlord. Transfer these <bunks> slaves in secure cells to <destination> for <payment>. Slavery is punishable by death throughout the galaxy, so this may be a dangerous job to take."
 	passengers 10 4 .1
+	# TODO: illegal -1 should be an atrocity, but it isn't
```
```suggestion
```

warp-core commented on data/human/pirate jobs.txt:
```diff
@@ -1765,6 +1789,7 @@ mission "Bulk Slave Transport [1]"
 	repeat
 	description "A pirate on <planet> has bought slaves from a local warlord. Transfer these <bunks> slaves in secure cells to <destination> for <payment>. Slavery is punishable by death throughout the galaxy, so this may be a dangerous job to take."
 	passengers 10 4 .1
+	# TODO: illegal -1 should be an atrocity, but it isn't
```
```suggestion
```

ziproot commented:
> This really isn't the place for a comment like this.
> If you believe your syntax is in line with the intended design, and for whatever reason, the game doesn't handle it as intended, open an issue.

My bad. I will avoid doing this in the future.

> I suspect the problem here is not that "illegal -1" isn't counted as an atrocity.
> Instead, I believe the issue is that missions with illegal passengers are not handled properly.
> When "EvadesCargoScan" from the anonymous namespace in Politics.cpp checks how much illegal cargo the ship has, it calls a method of "CargoHold" that does not consider passengers.
> I expect a positive "illegal" value here (representing a fine) will also not work correctly.

A positive illegal value works, as that is what is used in the other pirate missions, where I am fined. The issue is that this wasn’t always the case, and when Amazinite made it so that illegal attributes didn’t have to be child nodes for cargo nodes, this was not done for atrocities.

warp-core commented:
> > I suspect the problem here is not that "illegal -1" isn't counted as an atrocity.
> > Instead, I believe the issue is that missions with illegal passengers are not handled properly.
> > When "EvadesCargoScan" from the anonymous namespace in Politics.cpp checks how much illegal cargo the ship has, it calls a method of "CargoHold" that does not consider passengers.
> > I expect a positive "illegal" value here (representing a fine) will also not work correctly.
> 
> A positive illegal value works, as that is what is used in the other pirate missions, where I am fined. The issue is that this wasn’t always the case, and when Amazinite made it so that illegal attributes didn’t have to be child nodes for cargo nodes, this was not done for atrocities.

I'm fairly sure those other missions have cargo, not just passengers.

> when Amazinite made it so that illegal attributes didn’t have to be child nodes for cargo nodes, this was not done for atrocities.

The "illegal" property was moved from being a child of "cargo" to a child of "mission" here: https://github.com/endless-sky/endless-sky/commit/03b24218338c5377d6a0849a58e2b3a52498b75f#diff-327f44680dd49f3ee9b019252c27e2de286c57ec7fe66321224125bbeb01e1c8

ziproot commented:
Oh right, my bad, [Amazinite updated the pirate jobs](https://github.com/endless-sky/endless-sky/commit/a86c195e559195986521be6f4a27aa1d6cd8e2df), and only the slavery missions. 

I just tested with the shady passenger mission, and was not caught despite being scanned by Hai ships and landing on Everhope, which has a security of 1. I forgot that all of the other pirate jobs had cargo.

warp-core reviewed (commented):
What's the reason for the increase in minimum distance on these?

I don't think the other ships alongside Ryk Bartlett should be "marked". I don't think they're supposed to be the main focus of the mission, but making them "marked" means they will all rush for the player. It doesn't just prevent other ships from not attacking them, it makes them only consider attacking the player as well.

I'd like the change to the credits to be separated from the other changes here. Either open a new PR for it or it can be handled when we update the credits and stuff for 0.10.7 next month. (I'm not really sure the changes to the Free Worlds mission should be alongside the changes to all these jobs, either.)

ziproot commented:
> What's the reason for the increase in minimum distance on these?

This stops CCOR planets from offering smuggling missions to Men. I still want missions that can go there, but not if they only go through pirate space.

ziproot commented:
> I don't think the other ships alongside Ryk Bartlett should be "marked". I don't think they're supposed to be the main focus of the mission, but making them "marked" means they will all rush for the player. It doesn't just prevent other ships from not attacking them, it makes them only consider attacking the player as well.

I made them the bounty government in #6543, but Amazinite rejected that. I mentioned in that pull request that it meant other friendly ships could do the mission for you. I guess the support ships being target-able by friendly ships is fine, if the main ship still has to be attacked by you.

warp-core commented on data/human/pirate jobs.txt:
```diff
@@ -2137,7 +2162,9 @@ mission "Escort Illegal Cargo (Large, South, Dangerous Path)"
 		government "Pirate"
 		attributes "south pirate"
-		distance 3 6
+		# militia wouldn't be here
+		not attributes ccor
```
How about requiring the system have a non-Pirate neighbour?
```suggestion
		neighbor
			not government "Pirate"
```
or
```suggestion
		neighbor
			government "Republic" "Free Worlds" "Neutral"
```

warp-core commented on data/human/pirate jobs.txt:
```diff
@@ -2332,11 +2361,13 @@ mission "Eliminating Law Enforcement (South, Small)"
+		personality heroic marked staying target
 		system
 			distance 1 2
+			not attributes ccor
```
```suggestion
			not government "Pirate"
```
or
```suggestion
			government "Republic" "Free Worlds"
```

warp-core commented on data/human/pirate jobs.txt:
```diff
@@ -2683,6 +2720,8 @@ mission "Raid on Merchants (South)"
 		government "Merchant"
 		system
 			distance 1 3
+			neighbor
+				not government "Pirate"
```
Would it make sense to apply this to the North and Core versions of this job as well?
Currently, it seems like you could take the job on Freedom, and have the targets spawn in Alnilam, or on Stormhold and have targets in Atik, which probably shouldn't be the case.

ziproot replied:
Don’t navy ships patrol there?

ziproot replied:
> Also, why such a large reputation penalty with pirates in the slave jobs? The next highest penalty I see for failing a pirate job is 3.

That would also be a jump from -5 penalty to an atrocity for the republic/syndicate/free worlds, which sets reputation to minimum 0 and then decreases it by 10, however, I think I can probably make it -9 instead.

warp-core reviewed (commented):
It looks like you missed adding the "marked" personality to the targets in "Eliminating Law Enforcement (North, Small)".

Also, why such a large reputation penalty with pirates in the slave jobs? The next highest penalty I see for ffailing a pirate job is 3.
